check_net.sh

```diff
@@ -11,12 +11,12 @@ while true;
 do
 	date="[$(date +'%b %d, %r')] ";
 	works='true';
-	for check_address in "${ip_addresses[@]}"
+	for check_address in ${ip_addresses[@]}
 	do
 		if ping -W 5 -c 1 $check_address > /dev/null;
 		then
 			works='true';
-			break;
+			continue;
```

Comment:
No! :(
I think you miss the flow here... 
if 1 IP works, we have the internet and check in $sleep_cycle seconds. 
If it's `continue` than result ($works) will be only last IP of $ip_addresses